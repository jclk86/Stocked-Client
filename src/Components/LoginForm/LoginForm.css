.header_login {
  text-align: center;
  padding-top: 50px;
  margin-bottom: 20px;
  max-width: 100%;
}

.loading_gif {
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  margin: auto;
}

.logo_login {
  max-width: 150px;
}

.label_login {
  display: block;
  width: 85%;
  margin: 0 auto;
}

.login_register_container {
  text-align: center;
}

.alert_login {
  text-align: center;
}

.message_redirect {
  font-size: 16px;
}

.btn_register_link {
  text-decoration: none;
  color: rgba(46, 211, 189, 0.973);
  font-weight: bold;
}
