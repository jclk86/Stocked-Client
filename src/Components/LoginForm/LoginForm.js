import React, { Component } from "react";
import { Button, Input, Form, Logo } from "../Utils/Utils";
import { Link } from "react-router-dom";
import "./LoginForm.css";

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  };

  state = { error: null };

  render() {
    return (
      <Form>
        <div className="header_login">
          <Logo></Logo>
          <h2>STOCKED</h2>
        </div>
        <div className="user_name">
          <label htmlFor="LoginForm__user_name" className="label_login">
            User name
          </label>
          <Input required name="user_name" id="LoginForm__user_name" />
        </div>
        <div className="password">
          <label htmlFor="LoginForm__password" className="label_login">
            Password
          </label>
          <Input
            required
            name="password"
            type="password"
            id="LoginForm__password"
          />
        </div>
        <div className="login_register_container">
          <Button role="button" type="submit">
            Login
          </Button>
          <p>
            Not a member? <Link to="/register">Register here.</Link>
          </p>
        </div>
      </Form>
    );
  }
}
