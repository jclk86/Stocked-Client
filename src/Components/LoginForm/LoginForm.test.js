import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import LoginForm from "./LoginForm";
import sinon from "sinon";
import AuthApiService from "../../services/auth-api-service";

describe(`LoginForm Component`, () => {
  let mockData = {
    username: "Pop123",
    password: "Password123!",
    fullname: "Bob Smith",
    email: "BSmith@gmail.com"
  };

  beforeAll(() => {
    AuthApiService.postUser(mockData).then(user => {
      AuthApiService.postLogin({
        username: user.username,
        password: user.password
      });
    });
  });

  it("Renders an LoginForm without issue", () => {
    const wrapper = shallow(<LoginForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("fires submit", () => {
    const handleSubmitSpy = sinon.spy();
    const wrapper = mount(
      <BrowserRouter>
        <LoginForm onSubmit={handleSubmitSpy}></LoginForm>
      </BrowserRouter>
    );
    wrapper.find(".Form").simulate("submit", {
      target: { username: mockData.username, password: mockData.password },
      preventDefault: () => {}
    });
  });
});
