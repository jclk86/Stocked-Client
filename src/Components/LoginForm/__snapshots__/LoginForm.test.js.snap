// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component Renders an LoginForm without issue 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
